deep_field_metadetect/jaxify/jax_detection.py

```diff
@@ -0,0 +1,446 @@
+from functools import partial
+from typing import Tuple, Union
+
+import jax
+import jax.numpy as jnp
+
+
+@partial(jax.jit, static_argnames=["window_size"])
+def local_maxima_filter(
+    image: jnp.ndarray,
+    noise: Union[jnp.ndarray, float],
+    window_size: int = 5,
+) -> jnp.ndarray:
+    """
+    Find local maximas in an image within window_size
+
+    Parameters:
+    -----------
+    image : jnp.ndarray
+        2D Input galaxy field
+    window_size : int
+        Size of the neighborhood for local maximum detection
+    noise : jnp.ndarray | float
+        Pixelwise noise sigma
+        Minimum pixel value of central pixel must > 3-sigma
+
+    Returns:
+    --------
+    jnp.ndarray
+        Binary mask indicating local maxima positions
+    """
+    noise_array = jnp.broadcast_to(noise, image.shape) if jnp.isscalar(noise) else noise
```

Comment:
I suspect this if statement will be buggy under JIT but maybe JAX re-JITs the function if noise is a scalar and so it works fine?

Reply:
Yes, this should be fine because JAX re-JITs functions when the shape/dtype of a non-static parameter changes. Adding a test just to be sure